pkg/vulnerabilityreport/ttl_report_test.go

```diff
@@ -0,0 +1,124 @@
+package vulnerabilityreport_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/trivy-operator/pkg/ext"
+	"github.com/aquasecurity/trivy-operator/pkg/operator/etc"
+	"github.com/aquasecurity/trivy-operator/pkg/trivyoperator"
+	"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport"
+)
+
+const (
+	vreport = "vreport"
+	ns      = "default"
+)
+
+func TestRegenerateReportIfExpired(t *testing.T) {
+	// clock
+	clock := ext.NewSystemClock()
+
+	// scheme
+	scheme := trivyoperator.NewScheme()
+
+	// set the VulnerabilityScannerReportTTL
+	config, err := etc.GetOperatorConfig()
+	require.NoError(t, err)
+	hours, err := time.ParseDuration("24h")
+	require.NoError(t, err)
+	config.VulnerabilityScannerReportTTL = &hours
+
+	// logger object
+	logger := log.Log.WithName("testing")
+
+	// create TTLReport controller
+	instance := vulnerabilityreport.TTLReportReconciler{Logger: logger, Config: config, Clock: clock}
+
+	tests := []struct {
+		name                  string
+		reportUpdateTimestamp time.Duration
+		ttlAnnotationMissing  bool
```

Comment:
I think you can get rid of this by treating the empty `ttlStr` as this case.

Reply:
this flag removes the entire annotation when set, so helpful in testing missing entire annotation and invalid duration value

Reply:
I think you can use the same conditional with `ttlStr != ""`.

Reply:
removed